test/unit/org/apache/cassandra/index/sai/metrics/TinySegmentFlushingFailureTest.java

```diff
@@ -17,22 +17,65 @@
  */
 package org.apache.cassandra.index.sai.metrics;
 
-import org.junit.Before;
+import org.apache.cassandra.index.sai.disk.v1.V1OnDiskFormat;
 import org.junit.BeforeClass;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
 
+import java.lang.reflect.Field;
+
+import org.apache.cassandra.config.Config;
 import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.dht.Murmur3Partitioner;
+import org.apache.cassandra.service.StorageService;
+
+import static org.junit.Assert.assertEquals;
 
-public class TinySegmentFlushingFailureTest extends SegmentFlushingFailureTest
+@RunWith(Enclosed.class)
+public class TinySegmentFlushingFailureTest extends CQLTester
 {
-    @Before
-    public void setSegmentBufferSpace() throws Throwable
+    /**
+     * Set the necessary configuration before any tests run.
+     */
+    @BeforeClass
+    public static void setUpClass()
     {
-        setSegmentWriteBufferSpace(0);
+        try
+        {
+            Field confField = DatabaseDescriptor.class.getDeclaredField("conf");
+            confField.setAccessible(true);
+            Config conf = (Config) confField.get(null);
+            conf.sai_options.segment_write_buffer_space_mb = 0;
+
+            System.out.println("Configuration set: segment_write_buffer_space_mb = " + 0);
+        }
+        catch (NoSuchFieldException | IllegalAccessException e)
+        {
+            throw new RuntimeException("Failed to set configuration segment_write_buffer_space_mb = 0", e);
+        }
+        
+        if (ROW_CACHE_SIZE_IN_MB > 0)
```

Comment:
Could we not call `CQLTester.setUpClass()` here too?

Reply:
We can and we should, leftover from some testing. Fixing it